cl_dll/hud.cpp

```diff
@@ -301,6 +301,38 @@ void __CmdFunc_Append()
 	EngineClientCmd(gEngfuncs.Cmd_Argv(1));
 }
 
+void __CmdFunc_Savemap()
+{
+	FILE* saved_maps;
+	char map_name[64];
+	char map_name_to_check[64];
+
+	get_map_name(map_name, ARRAYSIZE(map_name));
+
+	if (map_name[0])
+	{
+		sprintf(map_name, "%s\n", map_name);
```

Comment:
If the map is 63 characters long + null byte, this will write one byte past the end of the array. I suggest comparing without `\n` below rather than appending `\n` here.